### PR TITLE
Trio support!

### PR DESCRIPTION
**Pull request by jfhbrook**
This PR adds trio support to pyee.

@decentral1se - does this look like it would work for your needs?

### CODE REVIEW

**Comment by jfhbrook** [comment]
- [ ] TODO: Update the docs

File: pyee/_trio.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+from contextlib import asynccontextmanager
+import trio
+from pyee._base import BaseEventEmitter, PyeeException
+
+__all__ = ['TrioEventEmitter']
+
+
+class TrioEventEmitter(BaseEventEmitter):
+    """An even emitter class which can run trio coroutines in a trio nursery.
```
Comment (decentral1se) [inline pyee/_trio.py]:
```suggestion
    """An event emitter class which can run trio coroutines in a trio nursery.
```

Comment (decentral1se) [inline pyee/_trio.py]:
```suggestion
    """An even emitter class which can run trio tasks in a trio nursery.
```

IIUC, tasks are the idea: https://trio.readthedocs.io/en/stable/design.html#basic-principles.

File: pyee/_trio.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    Unlike the case with the BaseEventEmitter, all exceptions raised by event
+    handlers are automatically emitted on the ``error`` event. This is
+    important for trio coroutines specifically but is also handled for
+    synchronous functions for consistency.
+
+    For trio coroutine event handlers, calling emit is non-blocking. In other
+    words, you should not attempt to await emit; the coroutine is scheduled
+    in a fire-and-forget fashion.
+
+    Note that this behavior is a little against the grain for typical trio use.
```
Comment (decentral1se) [inline pyee/_trio.py]: Because? Might be worth adding another sentence about callbacks and design etc.

File: pyee/_trio.py (hunk 1 above)
Comment (jfhbrook) [reply pyee/_trio.py]: Good call. Definitely want to use the UBIQUITOUS LANGUAGE here!

File: pyee/_trio.py (hunk 2 above)
Comment (jfhbrook) [reply pyee/_trio.py]: Yeah, low key thinking I should remove this comment and add a more broad disclaimer to the introduction in a separate PR.

**Comment by decentral1se** [comment]
:100:

**Pull request merged**